docs/notebooks/deep_gaussian_processes.pct.py

```diff
@@ -0,0 +1,355 @@
+# %% [markdown]
+# # Using deep Gaussian processes with GPflux for Bayesian optimization.
+
+# %%
+import numpy as np
+import tensorflow as tf
+
+np.random.seed(1794)
+tf.random.set_seed(1794)
+tf.keras.backend.set_floatx("float64")
+
+# %% [markdown]
+# ## Describe the problem
+#
+# In this notebook, we show how to use deep Gaussian processes (DGPs) for Bayesian optimization using Trieste and GPflux.
```

Comment:
it would be good to add a sentence on why this might be a good idea (nonstationary functions...) and point to a reference (you'll need to add it to the references file as well)